### PR TITLE
Adds tests

### PR DESCRIPTION
**Pull request by dainis**
Adds testing infrastructure (#36) 

Tests aren't passing right now,  porting @hallvors python tests and running them showed that both engines produce different results( [windowOrientationTest.js#L35-L46](https://github.com/Asynchq/boar/blob/test_setup/tests/integration/windowOrientationTest.js#L35-L46 ) fails on gecko but works on webkit and [loggingTest.js](https://github.com/Asynchq/boar/blob/test_setup/tests/integration/loggingTest.js) gecko will return line number but webkit won't)

### CODE REVIEW

**Comment by ronoc** [comment]
fixes #36

**Comment by seiflotfy** [comment]
both engines have different return structures. I can replicate windowOritenationTest not working on gecko at all... Is there another way to test the window orientation that is cross browser compatible?

**Comment by ronoc** [comment]
@hallvors ^

File: tests/integration/loggingTest.js [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+        return client.getErrorLog();
+      })
+      .then((data) => {
+        data.consoleLog.length.should.be.equal(0);
+        done();
+      })
+      .catch(done);
+    });
+
+    it('should be returning some output if there is compile time error', (done) => {
```
Comment (hallvors) [inline tests/integration/loggingTest.js]: Nitpick: this test is a runtime error, the one below is a compile time error. The descriptions got mixed up ;)

**Comment by hallvors** [comment]
@dainis: good work! :)
Would be nice if we can just work around the logging issue somewhere in the API, probably best done in the Boar code when responding to getConsoleLog / getErrorLog requests. If there is no way to get the line number in WebKit's implementation let's just drop it, it's not critically important although it would be nice to have.

I don't see any obvious reason why the orientation test fails, will look into it..

**Comment by hallvors** [comment]
(On Travis all the getConsoleLog tests seem to fail on WebKit - not sure why??)

**Comment by hallvors** [comment]
If you run the tests after rebasing to include this: https://github.com/Asynchq/boar/issues/43 - does the behaviour of the window.orientation test change?

**Comment by ronoc** [comment]
@hallvors, yep those tests now pass, still failing on the console error discrepancies though.

**Comment by hallvors** [comment]
:+1: - I'll look at the console stuff shortly, I have a computer to test on handy now

**Comment by hallvors** [comment]
@dainis good if you also could switch the "runtime" and "compile-time" descriptions as noted above, thanks :)

**Comment by seiflotfy** [comment]
One of the following is our only way to solve this.
1) Use something prior to phantom 1.8.0 https://github.com/ariya/phantomjs/issues/10957
2) stop checking for lines  from the test data.consoleLog[0].lineNum.should.be.equal(1);
3) find a way to use phantom 2.0 (which would make Mac people not being able to use it)

**Comment by hallvors** [comment]
Drop the line number check. Not that important.

**Comment by seiflotfy** [comment]
fingers crossed

**Comment by hallvors** [comment]
Tests where green and the whole thing seemed so good I just merged it, hope you didn't mind :)

**Pull request merged**